tests/lib/test_phase_stats.py

```diff
@@ -278,8 +279,126 @@ def test_phase_stats_single_network_procfs():
+    assert network_energy_entry['phase'] == '004_[RUNTIME]'
+
+    network_totals = DB().fetch_all(
+        'SELECT detail_name, value FROM phase_stats WHERE phase = %s AND metric = %s',
+        params=('004_[RUNTIME]', 'network_total_cgroup_container'), fetch_mode='dict'
+    )
+    total_network_bytes = sum(row['value'] for row in network_totals)
+    expected_network_energy_kwh = Decimal(total_network_bytes) / 1_000_000_000 * Decimal(test_sci_config['N'])
+    expected_network_energy_uj = expected_network_energy_kwh * 3_600_000_000_000
+    assert abs(network_energy_entry['value'] - expected_network_energy_uj) < 1, \
+        f"Expected network energy: {expected_network_energy_uj}, got: {network_energy_entry['value']}"
+
+    # Network carbon data
+    network_carbon_data = DB().fetch_all(
+        'SELECT metric, detail_name, unit, value, type FROM phase_stats WHERE phase = %s AND metric = %s',
+        params=('004_[RUNTIME]', 'network_carbon_formula_global'), fetch_mode='dict'
+    )
+
+    assert len(network_carbon_data) == 1, "Expected 1 network carbon formula entry"
+
+    network_carbon_entry = network_carbon_data[0]
+    expected_network_carbon_ug = expected_network_energy_kwh * Decimal(test_sci_config['I']) * 1_000_000
+
+    assert network_carbon_entry['metric'] == 'network_carbon_formula_global'
+    assert network_carbon_entry['detail_name'] == '[FORMULA]'
+    assert network_carbon_entry['unit'] == 'ug'
+    assert network_carbon_entry['type'] == 'TOTAL'
+    assert abs(network_carbon_entry['value'] - expected_network_carbon_ug) < 1, \
+        f"Expected network carbon: {expected_network_carbon_ug}, got: {network_carbon_entry['value']}"
+
+def test_sci_calculation():
+    run_id = Tests.insert_run()
+    Tests.import_machine_energy(run_id)  # Machine energy component
+    Tests.import_network_io_cgroup_container(run_id)  # Network component (custom N parameter)
+
+    # Define comprehensive SCI configuration with all required parameters
+    test_sci_config = {
+        'N': 0.001,    # Network energy intensity (kWh/GB)
+        'I': 500,      # Carbon intensity (gCO2e/kWh)
+        'EL': 4,       # Expected lifespan (years)
+        'TE': 300000,  # Total embodied emissions (gCO2e)
+        'RS': 1,       # Resource share (100%)
+        'R': 10,       # Functional unit count (10 runs)
+        'R_d': 'test runs'  # Functional unit description
+    }
+
+    build_and_store_phase_stats(run_id, sci=test_sci_config)
+
+    # Verify all SCI components are calculated and stored correctly
+
+    # 1. Machine carbon from energy consumption
+    machine_carbon_data = DB().fetch_all(
+        'SELECT metric, value, unit FROM phase_stats WHERE phase = %s AND metric LIKE %s',
+        params=('004_[RUNTIME]', '%_carbon_ac_%'), fetch_mode='dict'
+    )
+    assert len(machine_carbon_data) >= 1, "Machine carbon should be calculated"
+    machine_carbon_ug = sum(row['value'] for row in machine_carbon_data if row['unit'] == 'ug')
```

Comment:
why filter here for unit? why not harden the test and fail when other unit occurs?